api/db/changelog/db.changelog-116-add-billing-account-to-workspaces.xml

```diff
@@ -0,0 +1,18 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<databaseChangeLog
+    xmlns="http://www.liquibase.org/xml/ns/dbchangelog/1.9"
+    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+    xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog/1.9
+                    http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-1.9.xsd">
+  <changeSet author="ericsong" id="db.changelog-116-add-billing-account-to-workspaces">
+    <addColumn tableName="workspace">
+      <column name="billing_account_name" type="varchar(100)">
+        <constraints nullable="false"/>
+      </column>
+    </addColumn>
+
+    <sql>update workspace set billing_account_name=(SELECT JSON_UNQUOTE(JSON_EXTRACT(configuration, '$.billing.accountId')) from config where config_id='main');</sql>
+
+    <dropColumn tableName="workspace" columnName="billing_account_type"/>
```

Comment:
I misunderstood, I thought the proposal was null (free tier) OR user_billing_account_name. I think this would be preferred as it makes the usage of free tier a more explicit, separate state and avoids reencoding the exact free tier billing name for each workspace.

Reply:
I was planning to add the free tier billing name for all the workspaces with a backfill because it simplifies the update/syncing code. 

Ex. in my feature branch, there is no need for special handling of a null or empty string in the billing_account_name field. The UI is able to match the value from `billing_account_name` to the accounts in `listBillingAccounts` for both the free tier and user supplied case. The same should be true for the update billing account case.

Given that, is it still worth keeping both fields? If so, I can add a constraint to ensure that the type is always set to free tier if the billing account name matches the config account name.

Reply:
+1 for keeping both fields. It seems fragile to rely on a naming convention, especially as project names aren't enforced by the DB schema. Also, how is the local vs test env name handled? Are they the same billing acct?

Reply:
Having both columns sounds fine. I would not add any constraints that created dependencies between the config table and the backend data - I think we need to allow these to remain independent; the fact that we store our configs in SQL is mostly out of convenience rather than any logical requirement, and keeping them in lock-step may not always be correct (for example, consider projects pending assignment in the buffer).

I'm happy with either:
1. a billingAccountName and a billingType column, or
2. just userBillingAccountName (null if free tier)

@jaycarlton there is no proposed dependency here on naming conventions. local and test are the same account, the value is in the config JSONs if you want to reference it.

Reply:
Thanks. I misunderstood. I was thinking one proposal involved inferring something from a name that was outside the database's control. Whether "not null" is a naming convention or not... 😉 

It's still a bit odd to me that we have baked in an assumption that the presence of an account always indicates paid tier and vice versa. It seems like the kind of thing that reality might make less convenient. 

Option #1 seems like it's clearer in the short term and gives more flexibility in the long term (exactly what CH said in the pending case).